src/ValueParsers/QuantityParser.php

```diff
@@ -88,18 +88,23 @@ protected function stringParse( $value ) {
 	 * @return QuantityValue
 	 */
 	private function newQuantityFromParts( $amount, $exactness, $margin, $unit ) {
+		list( $amount, $exponent ) = $this->decimalParser->splitDecimalExponent( $amount );
 		$amountValue = $this->decimalParser->parse( $amount );
```

Comment:
This is a bit odd because it means splitDecimalExponent is called twice. I'm pretty sure QuantityParser will now accept stuff like `12e3e4`. It obviously shouldn't, right?